isis/src/base/objs/SpicePosition/SpicePosition.cpp

```diff
@@ -1305,8 +1327,37 @@ namespace Isis {
         p_velocity[1] = (p2[1] - p1[1]) * mult + p1[1];
         p_velocity[2] = (p2[2] - p1[2]) * mult + p1[2];
       }
+    }*/
+    if (p_hasVelocity) {
+      std::vector<ale::State> stateCache;
+      for (int i=0; i < p_cache.size(); i++) {
+        stateCache.push_back(ale::State(p_cache[i], p_cacheVelocity[i]));
+      }
+      ale::States states(p_cacheTime, stateCache);
```

Comment:
It is really in-efficient to re-create this every time this gets called. You should just replace p_cache with a States object.